src/Stream.php

```diff
@@ -676,6 +712,71 @@ public function toArray(): array
         return $result;
     }
 
+    /**
+     * Writes iterable source to the file line by line.
+     *
+     * Elements of the iterable source must be stringifiable.
+     *
+     * @param resource $file
+     * @param string $separator (optional) inserted between each item. Ex: ', ' for 1, 2, 3, ...
+     * @param string $header    (optional) prepended to string
+     * @param string $footer    (optional) appended to string
+     *
+     * @return void
+     */
+    public function toFile($file, string $separator = \PHP_EOL, string $header = '', string $footer = ''): void
+    {
+        ResourceHelper::checkIsValid($file);
+
+        $firstIteration = true;
+
+        fputs($file, $header);
+
+        foreach ($this->iterable as $line) {
+            if ($firstIteration) {
+                $firstIteration = false;
+            } else {
+                $line = $separator . \strval($line);
+            }
+
+            fputs($file, \strval($line));
+        }
+
+        fputs($file, $footer);
```

Comment:
The newline separator would have to be added after the header if one is provided, and before the footer if one is provided, otherwise it isn't functioning as a header and footer. So I think you'd need some if statement prior to each to determine if you need to write them and put in the separator. Since empty line would be a valid header (and footer), perhaps the default value for header/footer should be null, then the if statement is if it is not null, add the header + newline separator, and similarly for the footer.

For example, if I do something like this:
```php
$header = '<ul>';
$footer = '</ul>';

Stream::of(['item1', 'item2', 'item3'])
    ->map(fn ($item) => "  <li>$item</li>")
    ->toFile($fh, \PHP_EOL, $header, $footer);
```

I'd expect the result to look like this:
```html
<ul>
  <li>item1</li>
  <li>item2</li>
  <li>item3</li>
</ul>
```

Reply:
Done.